include/core/SurfaceLayer.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef __NGEN_SURFACE_LAYER__
+#define __NGEN_SURFACE_LAYER__
+
+#include "Layer.hpp"
+
+namespace ngen
+{
+    class SurfaceLayer : public Layer
+    {
+        public:
+        
+        SurfaceLayer(
+                const LayerDescription& desc, 
+                const std::vector<std::string>& p_u, 
+                const Simulation_Time& s_t, 
+                feature_type& f, 
+                geojson::GeoJSON cd, 
+                std::shared_ptr<pdm03_struct> pdmd,
+                long idx,
+                const std::vector<std::string>& n_u,
+                std::unordered_map<std::string, std::ofstream>& output_files) : 
+                    Layer(desc,p_u,s_t,f,cd,pdmd,idx), 
+                    nexus_ids(n_u), 
+                    nexus_outfiles(output_files)
+        {
+
+        }
+
+        /***
+         * @breif Run one simulation timestep for each model in this layer
+        */
+        virtual void update_models()
+        {
+            long current_time_index = output_time_index;
+            
+            Layer::update_models();
+
+            //At this point, could make an internal routing pass, extracting flows from nexuses and routing
+            //across the flowpath to the next nexus.
+            //Once everything is updated for this timestep, dump the nexus output
+            for(const auto& id : features.nexuses()) 
+            {
+                std::string current_timestamp = simulation_time.get_timestamp(current_time_index);
```

Comment:
```suggestion
                if(!nexus_outfiles[id].is_open()) continue;

                std::string current_timestamp = simulation_time.get_timestamp(current_time_index);
```

Is `SurfaceLayer::update_models()` ever expected to do actual work, or is it purely here to handle the necessary output? In the latter case, we can omit all the other data juggling up front.